_posts/2023-02-13-Branch-Review-and-Learn.md

```diff
@@ -0,0 +1,135 @@
+
+__Please note:__ the content below represents one opinion in a larger space of Git workflow concepts (it's not perfect!). Developer cultures may vary on these topics; be sure to acknowledge people and culture over exclusive or absolute dedication to what is found below.
+
+__TLDR (too long, didn't read);__
+Use [git branching techniques](https://www.atlassian.com/git/tutorials/using-branches) to segment the completion of programming tasks, gradually and consistently committing small changes (practicing [_festina lente_ or "make haste, slowly"](https://en.wikipedia.org/wiki/Festina_lente)). When a group of small changes are ready from branches, request [pull request reviews](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) and take advantage of [comments](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request) to [continuously improve](https://en.wikipedia.org/wiki/Continual_improvement_process) the work. Prepare for a branch [merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/merging-a-pull-request) after review by deciding [which merge strategy](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/about-merge-methods-on-github) is appropriate and automating merge requirements with [branch protection rules](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches).
+
+## Concept: Coursework Branching
+
+<pre class="mermaid">
+flowchart LR
+ subgraph Course
+    direction LR
+    open["open\nassignment"]
+    turn_in["review\nassignment"]
+  end
+  subgraph Learner ["&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Learner"]
+    direction LR
+    work["completed\nassignment"]
+  end
+  open -.-> turn_in
+  open --> |works towards| work
+  work --> |seeks review| turn_in
+</pre>
+<script type="module">
+  import mermaid from 'https://unpkg.com/mermaid@9/dist/mermaid.esm.min.mjs';
+  mermaid.initialize({ startOnLoad: true });
+</script>
+
+_An example course and learner assignment workflow._
+{:.center}
+
+[Git branching](https://www.atlassian.com/git/tutorials/using-branches) practices may be understood in context with similar workflows from real life. Consider a scenario within an educational course where an assignment is made available to a learner. In addition to the steps shown in the diagram above, it's important to think about _why_ this pattern is beneficial:
```

Comment:
Is there a reason you're calling the person taking a course a "learner" rather than a "student"? I just feel like this text is unnecessarily formal for the intended audience, who I assume is lay people interested in using Git. I'd prefer the scenario to be explained in simpler terms, maybe something like "Consider a student taking a course, where an assignment is given to them to complete."

Reply:
Thanks, I see your point, and can move into student instead to simplify. I felt that the audience here might not be those interested in using git but perhaps those who are building software. Some of the content is built with the idea that they may need convincing to use additional features which may appear arbitrary or overly complex without more explanation. The "learner" use over "student" was to help avoid the connotation of hierarchy in the midst of parallels to a developer team (where there may not be a formal "teacher" so much as all being somewhat equally instructive).